src/main/java/rainier/org/csvcompare/compare/CsvCompareImp.java

```diff
@@ -0,0 +1,84 @@
+package rainier.org.csvcompare.compare;
+
+import de.siegmar.fastcsv.writer.CsvWriter;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.log4j.Logger;
+import rainier.org.csvcompare.config.Config;
+import rainier.org.csvcompare.dto.FileToCompare;
+
+public class CsvCompareImp implements CsvCompare<Map<String, String>> {
+
+  private static CsvCompareImp obj;
+
+  private FileToCompare<Map<String, String>> fileToCompare;
+
+  private CsvWriter csvWriter;
+
+  private static final Logger logger = Logger.getLogger(CsvCompareImp.class);
+
+  private CsvCompareImp() {
+    csvWriter = new CsvWriter();
+  }
+
+  public static CsvCompareImp getInstance() {
```

Comment:
singleton right. shouldn't his be synchronised, to make thread safe.

Reply:
hmmm, let's talk about this when you got a time.